factcast-factus/src/main/java/org/factcast/factus/projection/LocalWriteToken.java

```diff
@@ -27,7 +27,7 @@
 
   private final ReentrantLock lock = new ReentrantLock(true);
 
-  public AutoCloseable acquireWriteToken(@NonNull Duration maxWait) {
+  public WriterToken acquireWriteToken(@NonNull Duration maxWait) {
     try {
       if (lock.tryLock(maxWait.toMillis(), TimeUnit.MILLISECONDS)) {
         return lock::unlock;
```

Comment:
I would suggest to overwrite `isValid`, else someone could reuse a lock by accident:
```
return new WriterToken() {
 @Override
  public void close() throws Exception {
    lock.unlock();
  }    

  @Overwrite
  public boolean isValid() {
    return lock.isLocked();
  }
}
```